app/workers/certificates.php

```diff
@@ -26,169 +28,224 @@ public function init(): void
-      keyFile: /storage/certificates/{$domain->get()}/privkey.pem";
+            // Read new renew date from cert file
+            // TODO: This might not be required, we could calculate it. But this feels safer
+            $certPath = APP_STORAGE_CERTIFICATES . '/' . $domain->get() . '/cert.pem';
+            $certData = openssl_x509_parse(file_get_contents($certPath));
+            $validTo = $certData['validTo_time_t'];
+            $expiryInAdvance = (60*60*24*30);
+            $certificate->setAttribute('renewDate', $validTo - $expiryInAdvance);
+
+            // All went well at this point 🥳
+            
+            // Reset attempts count for next renwal
+            $certificate->setAttribute('attempts', 0);
+
+            // Mark issue date
+            $certificate->setAttribute('issueDate', \time());
+        } catch(ExceptionCertificate $e) {
+            // These exceptions are expected if renew shouldn't or can't happen
+
+            // Add exception as log into certificate
+            $certificate->setAttribute('log', $e->getMessage());
+
+            $attempt = $certificate->getAttribute('attempts', 0);
+            $attempt++;
+            
+            // Save increased attempts count if requested by exception
+            if($e->getIncrementAttempts()) {
+                $certificate->setAttribute('attempts', $attempt);
+            }
 
-        if (!\file_put_contents(APP_STORAGE_CONFIG . '/' . $domain->get() . '.yml', $config)) {
-            throw new Exception('Failed to save SSL configuration');
-        }
+            Console::warning('Cannot renew domain (' . $domain->get() . ') on attempt no. ' . $attempt . ' certificate: ' . $e->getMessage());
+        } finally {
+            // All actions result in new updatedAt date
+            $certificate->setAttribute('updated', \time());
+
+            // Save certificate data into database
+            // Check if update or insert required
+            $certificateDocument = $dbForConsole->findOne('certificates', [ new Query('domain', Query::TYPE_EQUAL, [$domain->get()]) ]);
+            if (!empty($certificateDocument) && !$certificateDocument->isEmpty()) {
+                // Merge new data with current data
+                $certificate = new Document(\array_merge($certificateDocument->getArrayCopy(), $certificate->getArrayCopy()));
+                
+                $certificate = $dbForConsole->updateDocument('certificates', $certificate->getId(), $certificate);
+            } else {
+                $certificate = $dbForConsole->createDocument('certificates', $certificate);
+            }
 
-        ResqueScheduler::enqueueAt($renew + $safety, 'v1-certificates', 'CertificatesV1', [
-            'document' => [],
-            'domain' => $domain->get(),
-            'validateTarget' => $validateTarget,
-            'validateCNAME' => $validateCNAME,
-        ]);  // Async task rescheduale
+            // Update domains with new certificate ID
```

Comment:
Are we missing something here? why do we need `$certificateId`?

Reply:
Missing TODO. We need to update all domains to have `certificateId`. Required for proper UI functionality of custom domains.

Reply:
Implemented.